Rework dist modules to test against threaded builds of perl as well

tonycoz opened the pull request:
And fix some problems found,

tonycoz committed:
test-dist-modules.pl: make separate installation the default

This avoids the dangerous default of overwriting installed modules.

tonycoz committed:
testsuite.yml: test dist/ modules on threaded perls too

This originally also tested on Windows, but Storable fails a test on
5.24, and several other modules fail on 5.10 and 5.8.

tonycoz committed:
Devel::PPPort: add PERL_STACK_REALIGN

This was causing build failures when building threads.pm against older
perls.

tonycoz committed:
threads: thread_locale_init/term() needs 5.27.9 or later

tonycoz committed:
Storable: add test names to t/utf8.t

tonycoz committed:
Devel::PPPort: threads-shared needs SvREFCNT_dec_NN()

and threads.xs no longer needs to define SvREFCNT_dec_NN()

jkeenan commented:
Until reviewing this p.r., I was completely unaware of the existence of `Porting/test-dist-modules.pl`.  Is it meant to be run anywhere other than in our GH CI setups?  If so, how would I call it on my laptop and what results should I expect?

jkeenan commented:
> Until reviewing this p.r., I was completely unaware of the existence of `Porting/test-dist-modules.pl`. Is it meant to be run anywhere other than in our GH CI setups? If so, how would I call it on my laptop and what results should I expect?

@tonycoz, can you comment ^^?  thanks.

tonycoz commented:
> Until reviewing this p.r., I was completely unaware of the existence of `Porting/test-dist-modules.pl`. Is it meant to be run anywhere other than in our GH CI setups? If so, how would I call it on my laptop and what results should I expect?

I added it when I added dist-modules to the github workflows, it found a few problems at the time.

It is primarily intended to do the CI checks. but it should work from your laptop.

it does require a clean build tree, it tests the `dist/` modules against the perl it was invoked with.

jkeenan commented on Porting/test-dist-modules.pl:
```diff
@@ -11,11 +11,16 @@
            "h|help"     => \&usage)
   or die "Unknown options\n";
 
+$separate
```
We could use some inline comments that describe what the various command-line options to `Porting/test-dist-modules.pl`.  And since we're looking at this statement, why does the program not exit once `&usage` is called (or when I call the program with `--help`?

tonycoz replied:
I've added POD and usage() now does an exit().

iabyn commented:
On Sat, Jul 13, 2024 at 12:46:03AM -0700, Tony Cook wrote:
> > Until reviewing this p.r., I was completely unaware of the existence of `Porting/test-dist-modules.pl`. Is it meant to be run anywhere other than in our GH CI setups? If so, how would I call it on my laptop and what results should I expect?
> 
> I added it when I added dist-modules to the github workflows, it found a few problems at the time.

It appears to be completely undocumented :-(
it would be nice to have, as a bare minimum, a few lines of comments at the
top of the file to explain its basic purpose, and what its args are.

-- 
Monto Blanco... scorchio!

tonycoz committed:
test-dist-modules.pl: update usage(), exit on usage

tonycoz committed:
test-dist-modules.pl: add some POD
